apps/server/src/modules/transfer/export/resource-exporter.ts

```diff
@@ -0,0 +1,38 @@
+import { injectable } from 'inversify';
+
+import { RESOURCE_TYPE } from '@flogo-web/plugins/flow-server';
+import {
+  ExportActionAgent,
+  Resource,
+  ResourceExportContext,
+} from '@flogo-web/lib-server/core';
+
+import { ResourceService } from '../../resources';
+import { AllContribsService } from '../../all-contribs';
+import { ResourcePluginRegistry } from '../../../extension';
+
+@injectable()
+export class ResourceExporter {
+  constructor(
+    private resourceService: ResourceService,
+    private pluginRegistry: ResourcePluginRegistry,
+    private allContribsService: AllContribsService
+  ) {}
+
+  async export(resourceId) {
+    const resource = await this.resourceService.findOne(resourceId);
+    const contributions = await this.allContribsService.allByRef();
+    const resourceTypes = this.pluginRegistry.resourceTypes;
+    const resourceExporter = resourceTypes.exporter(RESOURCE_TYPE);
+    const exportContext: ResourceExportContext = {
+      contributions,
+      resourceIdReconciler: new Map<string, Resource>(),
+      refAgent: {
+        getAliasRef: (contribType: string, ref: string) => ref,
+        registerFunctionName: () => {},
```

Comment:
I wonder that if we don't send the function imports to the runner service when executing a mapping expression that uses a function the service will error because it's not able to resolve the function? Or does it do some special magic to load all functions? @apandurangi

Reply:
It won't be a problem @fcastill . Those functions are already registered in the imports of our flogo-web local engine which is coordinating with the services for test running the flow.